lms/djangoapps/course_api/api.py

```diff
@@ -19,6 +19,8 @@ def get_effective_user(requesting_user, target_username):
     """
     if target_username == requesting_user.username:
         return requesting_user
+    elif target_username == '' and requesting_user.username:
```

Comment:
What is the rationale for the 2nd half of the `and` clause?  Is it really needed?